Exceptions when using current test library within spring zeebe

I create an in-memory engine in Spring Zeebe with the latest 1.3.2 version, when registering a worker I get this exception.

However, it also seems to be a timing issue of some sort, with re-arranging some spring code I also got around that exception. But I still did not figure out the exact why...

Here is the log - the sequence looks good to me:
```
2022-01-31 10:02:26.706  INFO 31280 --- [           main] s.c.c.ZeebeTestClientSpringConfiguration : Create Zeebe in-memory engine for test run...
2022-01-31 10:02:26.709  WARN 31280 --- [           main] io.camunda.zeebe.test.util               : No system property 'testForkNumber' set, using default value 0
2022-01-31 10:02:26.710  WARN 31280 --- [           main] io.camunda.zeebe.test.util               : No system property 'testMavenId' set, using default value 0
2022-01-31 10:02:26.710  INFO 31280 --- [           main] io.camunda.zeebe.test.util.SocketUtil    : Starting socket assignment with testForkNumber 0 and testMavenId 0
2022-01-31 10:02:26.712  INFO 31280 --- [           main] io.camunda.zeebe.test.util.SocketUtil    : Choosing next port 1025 for test fork 0 with range PortRange{host='localhost', basePort=1025, maxOffset=100, currentOffset=1}
2022-01-31 10:02:27.396  INFO 31280 --- [   -zb-actors-3] org.camunda.feel.FeelEngine              : Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.el.impl.feel.MessagePackValueMapper@42114dfe)), function-provider: io.camunda.zeebe.el.impl.feel.FeelFunctionProvider@23dfaf20, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@343f683a, configuration: Configuration(false)]
2022-01-31 10:02:27.500  INFO 31280 --- [   -zb-actors-3] io.camunda.zeebe.logstreams              : Recovered state of partition 1 from snapshot at position -1
2022-01-31 10:02:27.514  INFO 31280 --- [   -zb-actors-3] org.camunda.feel.FeelEngine              : Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.el.impl.feel.MessagePackValueMapper@982d5cd)), function-provider: io.camunda.zeebe.el.impl.feel.FeelFunctionProvider@db9df5b, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@11b8be9d, configuration: Configuration(false)]
2022-01-31 10:02:27.524  INFO 31280 --- [   -zb-actors-3] org.camunda.feel.FeelEngine              : Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.el.impl.feel.MessagePackValueMapper@392c9e40)), function-provider: io.camunda.zeebe.el.impl.feel.FeelFunctionProvider@22f1fa5e, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@59589caf, configuration: Configuration(false)]
2022-01-31 10:02:27.526  INFO 31280 --- [   -zb-actors-3] org.camunda.feel.FeelEngine              : Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.el.impl.feel.MessagePackValueMapper@d016d04)), function-provider: io.camunda.zeebe.el.impl.feel.FeelFunctionProvider@7dd981e6, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@12a39577, configuration: Configuration(false)]
2022-01-31 10:02:27.583  INFO 31280 --- [   -zb-actors-3] io.camunda.zeebe.processor               : Processor starts replay of events. [snapshot-position: -1, replay-mode: PROCESSING]
2022-01-31 10:02:27.583  INFO 31280 --- [           main] s.c.c.ZeebeTestClientSpringConfiguration : Started up Zeebe in-memory engine for test run
2022-01-31 10:02:27.585  INFO 31280 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'testInMemoryZeebeEngine' of type [io.camunda.zeebe.process.test.testengine.InMemoryEngineImpl] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2022-01-31 10:02:27.583  INFO 31280 --- [   -zb-actors-3] io.camunda.zeebe.processor               : Processor finished replay, with [lastProcessedPosition: -1, lastWrittenPosition: -1]
2022-01-31 10:02:27.587  INFO 31280 --- [   -zb-actors-3] io.camunda.zeebe.engine.state.migration  : Starting processing of migration tasks (use LogLevel.DEBUG for more details) ... 
2022-01-31 10:02:27.591  INFO 31280 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'testZeebeClientObjectFactory' of type [io.camunda.zeebe.spring.client.config.ZeebeTestClientSpringConfiguration$1] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2022-01-31 10:02:27.593  INFO 31280 --- [   -zb-actors-3] io.camunda.zeebe.engine.state.migration  : Skipping ProcessMessageSubscriptionSentTimeMigration migration (1/3).  It was determined it does not need to run right now.
2022-01-31 10:02:27.593  INFO 31280 --- [   -zb-actors-3] io.camunda.zeebe.engine.state.migration  : Skipping MessageSubscriptionSentTimeMigration migration (2/3).  It was determined it does not need to run right now.
2022-01-31 10:02:27.593  INFO 31280 --- [   -zb-actors-3] io.camunda.zeebe.engine.state.migration  : Skipping TemporaryVariableMigration migration (3/3).  It was determined it does not need to run right now.
2022-01-31 10:02:27.593  INFO 31280 --- [   -zb-actors-3] io.camunda.zeebe.engine.state.migration  : Completed processing of migration tasks (use LogLevel.DEBUG for more details) ... 
2022-01-31 10:02:27.600  INFO 31280 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'zeebeClientLifecycle' of type [io.camunda.zeebe.spring.client.ZeebeClientLifecycle] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2022-01-31 10:02:27.707  INFO 31280 --- [           main] i.c.z.s.client.config.JobHandlerTest     : Started JobHandlerTest in 1.937 seconds (JVM running for 3.417)
2022-01-31 10:02:27.718  INFO 31280 --- [           main] i.c.z.s.c.c.p.ZeebeWorkerPostProcessor   : Registering Zeebe worker(s) of bean: io.camunda.zeebe.spring.client.config.JobHandlerTest@be68757
2022-01-31 10:02:27.731  INFO 31280 --- [           main] i.c.z.s.c.c.p.ZeebeWorkerPostProcessor   : . Register Zeebe worker: ZeebeWorkerValue{name='test1', type='test1', timeout=-1, maxJobsActive=-1, requestTimeout=-1, pollInterval=-1, autoComplete=true, fetchVariables=[], methodInfo=io.camunda.zeebe.spring.client.bean.MethodInfo@7523a3dc}
2022-01-31 10:02:27.731  INFO 31280 --- [           main] i.c.z.s.c.c.p.ZeebeWorkerPostProcessor   : . Register Zeebe worker: ZeebeWorkerValue{name='io.camunda.zeebe.spring.client.config.JobHandlerTest.ORIGINAL#handleTest2', type='test2', timeout=-1, maxJobsActive=-1, requestTimeout=-1, pollInterval=-1, autoComplete=true, fetchVariables=[], methodInfo=io.camunda.zeebe.spring.client.bean.MethodInfo@2806d6da}
2022-01-31 10:02:28.133 ERROR 31280 --- [ault-executor-0] io.grpc.internal.SerializingExecutor     : Exception while executing runnable io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1HalfClosed@fe5ef2e

java.lang.IndexOutOfBoundsException: index=218 length=73 capacity=257
```

Full Stack Trace
```
2022-01-31 10:02:28.133 ERROR 31280 --- [ault-executor-0] io.grpc.internal.SerializingExecutor     : Exception while executing runnable io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1HalfClosed@fe5ef2e

java.lang.IndexOutOfBoundsException: index=218 length=73 capacity=257
	at org.agrona.concurrent.UnsafeBuffer.boundsCheck0(UnsafeBuffer.java:2221) ~[agrona-1.14.0.jar:1.14.0]
	at org.agrona.concurrent.UnsafeBuffer.putBytes(UnsafeBuffer.java:1135) ~[agrona-1.14.0.jar:1.14.0]
	at io.camunda.zeebe.msgpack.spec.MsgPackWriter.writeRaw(MsgPackWriter.java:132) ~[zeebe-msgpack-core-1.3.2.jar:1.3.2]
	at io.camunda.zeebe.msgpack.spec.MsgPackWriter.writeString(MsgPackWriter.java:144) ~[zeebe-msgpack-core-1.3.2.jar:1.3.2]
	at io.camunda.zeebe.msgpack.spec.MsgPackWriter.writeString(MsgPackWriter.java:139) ~[zeebe-msgpack-core-1.3.2.jar:1.3.2]
	at io.camunda.zeebe.msgpack.value.StringValue.write(StringValue.java:90) ~[zeebe-msgpack-value-1.3.2.jar:1.3.2]
	at io.camunda.zeebe.msgpack.property.BaseProperty.write(BaseProperty.java:91) ~[zeebe-msgpack-value-1.3.2.jar:1.3.2]
	at io.camunda.zeebe.msgpack.value.ObjectValue.write(ObjectValue.java:156) ~[zeebe-msgpack-value-1.3.2.jar:1.3.2]
	at io.camunda.zeebe.msgpack.value.ObjectValue.write(ObjectValue.java:82) ~[zeebe-msgpack-value-1.3.2.jar:1.3.2]
	at io.camunda.zeebe.msgpack.UnpackedObject.write(UnpackedObject.java:53) ~[zeebe-msgpack-value-1.3.2.jar:1.3.2]
	at io.camunda.zeebe.logstreams.impl.log.LogStreamWriterImpl.tryWrite(LogStreamWriterImpl.java:143) ~[zeebe-logstreams-1.3.2.jar:1.3.2]
	at io.camunda.zeebe.process.test.testengine.GrpcToLogStreamGateway.writeCommandWithoutKey(GrpcToLogStreamGateway.java:106) ~[zeebe-process-test-1.3.2.jar:1.3.2]
	at io.camunda.zeebe.process.test.testengine.GrpcToLogStreamGateway.activateJobs(GrpcToLogStreamGateway.java:297) ~[zeebe-process-test-1.3.2.jar:1.3.2]
	at io.camunda.zeebe.gateway.protocol.GatewayGrpc$MethodHandlers.invoke(GatewayGrpc.java:1472) ~[zeebe-gateway-protocol-impl-1.3.2.jar:1.3.2]
	at io.grpc.stub.ServerCalls$UnaryServerCallHandler$UnaryServerCallListener.onHalfClose(ServerCalls.java:182) ~[grpc-stub-1.43.0.jar:1.43.0]
	at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.halfClosed(ServerCallImpl.java:340) ~[grpc-core-1.43.0.jar:1.43.0]
	at io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1HalfClosed.runInContext(ServerImpl.java:866) ~[grpc-core-1.43.0.jar:1.43.0]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37) ~[grpc-core-1.43.0.jar:1.43.0]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133) ~[grpc-core-1.43.0.jar:1.43.0]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) ~[na:na]
	at java.base/java.lang.Thread.run(Thread.java:831) ~[na:na]

2022-01-31 10:02:28.133 ERROR 31280 --- [ault-executor-2] io.grpc.internal.SerializingExecutor     : Exception while executing runnable io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1HalfClosed@73da9c04

java.lang.IndexOutOfBoundsException: index=257 length=9 capacity=257
	at org.agrona.concurrent.UnsafeBuffer.boundsCheck0(UnsafeBuffer.java:2221) ~[agrona-1.14.0.jar:1.14.0]
	at org.agrona.concurrent.UnsafeBuffer.putBytes(UnsafeBuffer.java:1135) ~[agrona-1.14.0.jar:1.14.0]
	at io.camunda.zeebe.msgpack.spec.MsgPackWriter.writeRaw(MsgPackWriter.java:132) ~[zeebe-msgpack-core-1.3.2.jar:1.3.2]
	at io.camunda.zeebe.msgpack.spec.MsgPackWriter.writeString(MsgPackWriter.java:144) ~[zeebe-msgpack-core-1.3.2.jar:1.3.2]
	at io.camunda.zeebe.msgpack.spec.MsgPackWriter.writeString(MsgPackWriter.java:139) ~[zeebe-msgpack-core-1.3.2.jar:1.3.2]
	at io.camunda.zeebe.msgpack.value.StringValue.write(StringValue.java:90) ~[zeebe-msgpack-value-1.3.2.jar:1.3.2]
	at io.camunda.zeebe.msgpack.property.BaseProperty.write(BaseProperty.java:90) ~[zeebe-msgpack-value-1.3.2.jar:1.3.2]
	at io.camunda.zeebe.msgpack.value.ObjectValue.write(ObjectValue.java:156) ~[zeebe-msgpack-value-1.3.2.jar:1.3.2]
	at io.camunda.zeebe.msgpack.value.ObjectValue.write(ObjectValue.java:82) ~[zeebe-msgpack-value-1.3.2.jar:1.3.2]
	at io.camunda.zeebe.msgpack.UnpackedObject.write(UnpackedObject.java:53) ~[zeebe-msgpack-value-1.3.2.jar:1.3.2]
	at io.camunda.zeebe.logstreams.impl.log.LogStreamWriterImpl.tryWrite(LogStreamWriterImpl.java:143) ~[zeebe-logstreams-1.3.2.jar:1.3.2]
	at io.camunda.zeebe.process.test.testengine.GrpcToLogStreamGateway.writeCommandWithoutKey(GrpcToLogStreamGateway.java:106) ~[zeebe-process-test-1.3.2.jar:1.3.2]
	at io.camunda.zeebe.process.test.testengine.GrpcToLogStreamGateway.activateJobs(GrpcToLogStreamGateway.java:297) ~[zeebe-process-test-1.3.2.jar:1.3.2]
	at io.camunda.zeebe.gateway.protocol.GatewayGrpc$MethodHandlers.invoke(GatewayGrpc.java:1472) ~[zeebe-gateway-protocol-impl-1.3.2.jar:1.3.2]
	at io.grpc.stub.ServerCalls$UnaryServerCallHandler$UnaryServerCallListener.onHalfClose(ServerCalls.java:182) ~[grpc-stub-1.43.0.jar:1.43.0]
	at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.halfClosed(ServerCallImpl.java:340) ~[grpc-core-1.43.0.jar:1.43.0]
	at io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1HalfClosed.runInContext(ServerImpl.java:866) ~[grpc-core-1.43.0.jar:1.43.0]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37) ~[grpc-core-1.43.0.jar:1.43.0]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133) ~[grpc-core-1.43.0.jar:1.43.0]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) ~[na:na]
	at java.base/java.lang.Thread.run(Thread.java:831) ~[na:na]

2022-01-31 10:02:28.137  WARN 31280 --- [ault-executor-1] io.camunda.zeebe.client.job.poller       : Failed to activated jobs for worker test1 and job type test1

io.grpc.StatusRuntimeException: UNKNOWN
	at io.grpc.Status.asRuntimeException(Status.java:535) ~[grpc-api-1.43.0.jar:1.43.0]
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:479) ~[grpc-stub-1.43.0.jar:1.43.0]
	at io.grpc.internal.DelayedClientCall$DelayedListener$3.run(DelayedClientCall.java:463) ~[grpc-core-1.43.0.jar:1.43.0]
	at io.grpc.internal.DelayedClientCall$DelayedListener.delayOrExecute(DelayedClientCall.java:427) ~[grpc-core-1.43.0.jar:1.43.0]
	at io.grpc.internal.DelayedClientCall$DelayedListener.onClose(DelayedClientCall.java:460) ~[grpc-core-1.43.0.jar:1.43.0]
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:562) ~[grpc-core-1.43.0.jar:1.43.0]
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:70) ~[grpc-core-1.43.0.jar:1.43.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:743) ~[grpc-core-1.43.0.jar:1.43.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:722) ~[grpc-core-1.43.0.jar:1.43.0]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37) ~[grpc-core-1.43.0.jar:1.43.0]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133) ~[grpc-core-1.43.0.jar:1.43.0]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) ~[na:na]
	at java.base/java.lang.Thread.run(Thread.java:831) ~[na:na]

2022-01-31 10:02:28.137  WARN 31280 --- [ault-executor-3] io.camunda.zeebe.client.job.poller       : Failed to activated jobs for worker io.camunda.zeebe.spring.client.config.JobHandlerTest.ORIGINAL#handleTest2 and job type test2

io.grpc.StatusRuntimeException: UNKNOWN
	at io.grpc.Status.asRuntimeException(Status.java:535) ~[grpc-api-1.43.0.jar:1.43.0]
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:479) ~[grpc-stub-1.43.0.jar:1.43.0]
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:562) ~[grpc-core-1.43.0.jar:1.43.0]
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:70) ~[grpc-core-1.43.0.jar:1.43.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:743) ~[grpc-core-1.43.0.jar:1.43.0]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:722) ~[grpc-core-1.43.0.jar:1.43.0]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37) ~[grpc-core-1.43.0.jar:1.43.0]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:133) ~[grpc-core-1.43.0.jar:1.43.0]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) ~[na:na]
	at java.base/java.lang.Thread.run(Thread.java:831) ~[na:na]
```

In a different test (https://github.com/berndruecker/camunda-cloud-zeebe-tests-playgorund/blob/main/src/test/java/io/berndruecker/playground/zeebe/tests/SimpleSpringBootTest.java), I am seeing

```
10:13:08.553 [main] INFO  io.camunda.zeebe.spring.client.config.ZeebeTestClientSpringConfiguration - Create Zeebe in-memory engine for test run...
10:13:08.558 [main] WARN  io.camunda.zeebe.test.util - No system property 'testForkNumber' set, using default value 0
10:13:08.559 [main] WARN  io.camunda.zeebe.test.util - No system property 'testMavenId' set, using default value 0
10:13:08.559 [main] INFO  io.camunda.zeebe.test.util.SocketUtil - Starting socket assignment with testForkNumber 0 and testMavenId 0
10:13:08.560 [main] INFO  io.camunda.zeebe.test.util.SocketUtil - Choosing next port 1025 for test fork 0 with range PortRange{host='localhost', basePort=1025, maxOffset=100, currentOffset=1}
10:13:09.359 [main] INFO  io.camunda.zeebe.spring.client.config.ZeebeTestClientSpringConfiguration - Started up Zeebe in-memory engine for test run
10:13:09.361 [main] INFO  o.s.c.support.PostProcessorRegistrationDelegate$BeanPostProcessorChecker - Bean 'testInMemoryZeebeEngine' of type [io.camunda.zeebe.process.test.testengine.InMemoryEngineImpl] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
10:13:09.366 [main] INFO  o.s.c.support.PostProcessorRegistrationDelegate$BeanPostProcessorChecker - Bean 'testZeebeClientObjectFactory' of type [io.camunda.zeebe.spring.client.config.ZeebeTestClientSpringConfiguration$1] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
10:13:09.380 [main] INFO  o.s.c.support.PostProcessorRegistrationDelegate$BeanPostProcessorChecker - Bean 'zeebeClientLifecycle' of type [io.camunda.zeebe.spring.client.ZeebeClientLifecycle] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
10:13:09.413 [main] INFO  io.camunda.zeebe.spring.client.config.processor.DeploymentPostProcessor - deployment: ZeebeDeploymentValue{resources=[classpath*:*.bpmn], beanInfo=io.camunda.zeebe.spring.client.bean.ClassInfo@629984eb}
10:13:09.420 [main] INFO  io.camunda.zeebe.spring.client.config.processor.ZeebeWorkerPostProcessor - Registering Zeebe worker(s) of bean: io.berndruecker.playground.zeebe.tests.ServiceTaskAGlueCode@16da1abc
10:13:10.491 [main] INFO  io.camunda.zeebe.spring.client.config.processor.DeploymentPostProcessor - Deployed: <testProcess:1>
10:13:10.496 [main] INFO  io.camunda.zeebe.spring.client.config.processor.ZeebeWorkerPostProcessor - . Register Zeebe worker: ZeebeWorkerValue{name='serviceTaskAGlueCode#executeGlueCode', type='serviceTaskA', timeout=-1, maxJobsActive=-1, requestTimeout=-1, pollInterval=-1, autoComplete=true, fetchVariables=[a], methodInfo=io.camunda.zeebe.spring.client.bean.MethodInfo@4ed18798}
10:13:10.504 [main] INFO  io.berndruecker.playground.zeebe.tests.SimpleSpringBootTest - Started SimpleSpringBootTest in 2.937 seconds (JVM running for 4.28)
##############################################
##############################################
10:13:10.976 [grpc-default-executor-1] INFO  io.camunda.zeebe.client.job.poller - Activated 1 jobs for worker serviceTaskAGlueCode#executeGlueCode and job type serviceTaskA
 REST ENDPOINT INVOKED 
 ADAPTER INVOKED REST 
10:13:10.997 [-zb-actors-2] ERROR io.camunda.zeebe.processor - Expected to successfully process event 'LoggedEvent [type=0, version=0, streamId=1, position=31, key=2251799813685260, timestamp=1643620390990, sourceEventPosition=-1] RecordMetadata{recordType=COMMAND, intentValue=255, intent=ACTIVATE, requestStreamId=1, requestId=8, protocolVersion=3, valueType=JOB_BATCH, rejectionType=NULL_VAL, rejectionReason=, brokerVersion=1.3.2}' with processor, but caught an exception. Skip this event.
java.lang.RuntimeException: Could not deserialize object [JobBatchRecord]. Deserialization stuck at offset 91 of length 237
	at io.camunda.zeebe.msgpack.UnpackedObject.wrap(UnpackedObject.java:38)
	at io.camunda.zeebe.logstreams.impl.log.LoggedEventImpl.readValue(LoggedEventImpl.java:135)
	at io.camunda.zeebe.engine.processing.streamprocessor.RecordValues.readRecordValue(RecordValues.java:35)
	at io.camunda.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.processEvent(ProcessingStateMachine.java:243)
	at io.camunda.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.tryToReadNextEvent(ProcessingStateMachine.java:208)
	at io.camunda.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.readNextEvent(ProcessingStateMachine.java:189)
	at io.camunda.zeebe.util.sched.ActorJob.invoke(ActorJob.java:79)
	at io.camunda.zeebe.util.sched.ActorJob.execute(ActorJob.java:44)
	at io.camunda.zeebe.util.sched.ActorTask.execute(ActorTask.java:122)
	at io.camunda.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:97)
	at io.camunda.zeebe.util.sched.ActorThread.doWork(ActorThread.java:80)
	at io.camunda.zeebe.util.sched.ActorThread.run(ActorThread.java:189)
Caused by: java.lang.RuntimeException: Could not read property 'variables'
	at io.camunda.zeebe.msgpack.value.ObjectValue.read(ObjectValue.java:112)
	at io.camunda.zeebe.msgpack.UnpackedObject.wrap(UnpackedObject.java:32)
	... 11 common frames omitted
Caused by: io.camunda.zeebe.msgpack.spec.MsgpackReaderException: Unable to determine array type, found unknown header byte 0xc4 at reader offset 90
	at io.camunda.zeebe.msgpack.spec.MsgPackReader.exceptionOnUnknownHeader(MsgPackReader.java:474)
	at io.camunda.zeebe.msgpack.spec.MsgPackReader.readArrayHeader(MsgPackReader.java:107)
	at io.camunda.zeebe.msgpack.value.ArrayValue.read(ArrayValue.java:94)
	at io.camunda.zeebe.msgpack.property.BaseProperty.read(BaseProperty.java:75)
	at io.camunda.zeebe.msgpack.value.ObjectValue.read(ObjectValue.java:110)
	... 12 common frames omitted
10:13:11.014 [-zb-actors-2] ERROR io.camunda.zeebe.processor - Expected to process event 'TypedEventImpl{metadata=RecordMetadata{recordType=COMMAND, intentValue=255, intent=ACTIVATE, requestStreamId=1, requestId=8, protocolVersion=3, valueType=JOB_BATCH, rejectionType=NULL_VAL, rejectionReason=, brokerVersion=1.3.2}, value={"type":"","worker":"","timeout":-1,"maxJobsToActivate":-1,"jobKeys":[],"jobs":[],"variables":[],"truncated":false"deadline":[packed value (length=1)],"retries":[packed value (length=1)],"retryBackoff":[packed value (length=1)],"recurringTime":[packed value (length=1)],"customHeaders":[packed value (length=1)]}}' without errors, but exception occurred with message 'Could not deserialize object [JobBatchRecord]. Deserialization stuck at offset 91 of length 237'.
java.lang.RuntimeException: Could not deserialize object [JobBatchRecord]. Deserialization stuck at offset 91 of length 237
	at io.camunda.zeebe.msgpack.UnpackedObject.wrap(UnpackedObject.java:38)
	at io.camunda.zeebe.logstreams.impl.log.LoggedEventImpl.readValue(LoggedEventImpl.java:135)
	at io.camunda.zeebe.engine.processing.streamprocessor.RecordValues.readRecordValue(RecordValues.java:35)
	at io.camunda.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.processEvent(ProcessingStateMachine.java:243)
	at io.camunda.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.tryToReadNextEvent(ProcessingStateMachine.java:208)
	at io.camunda.zeebe.engine.processing.streamprocessor.ProcessingStateMachine.readNextEvent(ProcessingStateMachine.java:189)
	at io.camunda.zeebe.util.sched.ActorJob.invoke(ActorJob.java:79)
	at io.camunda.zeebe.util.sched.ActorJob.execute(ActorJob.java:44)
	at io.camunda.zeebe.util.sched.ActorTask.execute(ActorTask.java:122)
	at io.camunda.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:97)
	at io.camunda.zeebe.util.sched.ActorThread.doWork(ActorThread.java:80)
	at io.camunda.zeebe.util.sched.ActorThread.run(ActorThread.java:189)
Caused by: java.lang.RuntimeException: Could not read property 'variables'
	at io.camunda.zeebe.msgpack.value.ObjectValue.read(ObjectValue.java:112)
	at io.camunda.zeebe.msgpack.UnpackedObject.wrap(UnpackedObject.java:32)
	... 11 common frames omitted
Caused by: io.camunda.zeebe.msgpack.spec.MsgpackReaderException: Unable to determine array type, found unknown header byte 0xc4 at reader offset 90
	at io.camunda.zeebe.msgpack.spec.MsgPackReader.exceptionOnUnknownHeader(MsgPackReader.java:474)
	at io.camunda.zeebe.msgpack.spec.MsgPackReader.readArrayHeader(MsgPackReader.java:107)
	at io.camunda.zeebe.msgpack.value.ArrayValue.read(ArrayValue.java:94)
	at io.camunda.zeebe.msgpack.property.BaseProperty.read(BaseProperty.java:75)
	at io.camunda.zeebe.msgpack.value.ObjectValue.read(ObjectValue.java:110)
	... 12 common frames omitted

java.lang.RuntimeException: Could not deserialize object [JobBatchRecord]. Deserialization stuck at offset 91 of length 237

	at io.camunda.zeebe.msgpack.UnpackedObject.wrap(UnpackedObject.java:38)
	at io.camunda.zeebe.logstreams.impl.log.LoggedEventImpl.readValue(LoggedEventImpl.java:135)
	at io.camunda.zeebe.process.test.testengine.RecordStreamSourceImpl.mapToRecord(RecordStreamSourceImpl.java:171)
	at io.camunda.zeebe.process.test.testengine.RecordStreamSourceImpl.updateWithNewRecords(RecordStreamSourceImpl.java:149)
	at io.camunda.zeebe.process.test.testengine.RecordStreamSourceImpl.records(RecordStreamSourceImpl.java:46)
	at io.camunda.zeebe.process.test.testengine.RecordStreamSourceImpl.recordsOfValueType(RecordStreamSourceImpl.java:53)
	at io.camunda.zeebe.process.test.testengine.RecordStreamSourceImpl.processInstanceRecords(RecordStreamSourceImpl.java:60)
	at io.camunda.zeebe.process.test.filters.StreamFilter.processInstance(StreamFilter.java:9)
	at io.camunda.zeebe.process.test.assertions.ProcessInstanceAssert.isProcessInstanceCompleted(ProcessInstanceAssert.java:106)
	at io.camunda.zeebe.process.test.assertions.ProcessInstanceAssert.isCompleted(ProcessInstanceAssert.java:82)
	at io.berndruecker.playground.zeebe.tests.SimpleSpringBootTest.lambda$waitForCompletion$0(SimpleSpringBootTest.java:65)
	at org.testcontainers.shaded.org.awaitility.core.AssertionCondition.lambda$new$0(AssertionCondition.java:53)
	at org.testcontainers.shaded.org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:234)
	at org.testcontainers.shaded.org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:221)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:831)
Caused by: java.lang.RuntimeException: Could not read property 'variables'
	at io.camunda.zeebe.msgpack.value.ObjectValue.read(ObjectValue.java:112)
	at io.camunda.zeebe.msgpack.UnpackedObject.wrap(UnpackedObject.java:32)
	... 17 more
Caused by: io.camunda.zeebe.msgpack.spec.MsgpackReaderException: Unable to determine array type, found unknown header byte 0xc4 at reader offset 90
	at io.camunda.zeebe.msgpack.spec.MsgPackReader.exceptionOnUnknownHeader(MsgPackReader.java:474)
	at io.camunda.zeebe.msgpack.spec.MsgPackReader.readArrayHeader(MsgPackReader.java:107)
	at io.camunda.zeebe.msgpack.value.ArrayValue.read(ArrayValue.java:94)
	at io.camunda.zeebe.msgpack.property.BaseProperty.read(BaseProperty.java:75)
	at io.camunda.zeebe.msgpack.value.ObjectValue.read(ObjectValue.java:110)
	... 18 more
```

--- a/src/main/java/io/camunda/zeebe/process/test/testengine/GrpcToLogStreamGateway.java
+++ b/src/main/java/io/camunda/zeebe/process/test/testengine/GrpcToLogStreamGateway.java
@@ -94,20 +94,16 @@ public class GrpcToLogStreamGateway extends GatewayGrpc.GatewayImplBase implemen
 
   private void writeCommandWithKey(
       final Long key, final RecordMetadata metadata, final BufferWriter bufferWriter) {
-    synchronized (writer) {
-      writer.reset();
+    writer.reset();
 
-      writer.key(key).metadataWriter(metadata).valueWriter(bufferWriter).tryWrite();
-    }
+    writer.key(key).metadataWriter(metadata).valueWriter(bufferWriter).tryWrite();
   }
 
   private void writeCommandWithoutKey(
       final RecordMetadata metadata, final BufferWriter bufferWriter) {
-    synchronized (writer) {
-      writer.reset();
+    writer.reset();
 
-      writer.keyNull().metadataWriter(metadata).valueWriter(bufferWriter).tryWrite();
-    }
+    writer.keyNull().metadataWriter(metadata).valueWriter(bufferWriter).tryWrite();
   }
 
   @Override
@@ -284,21 +280,24 @@ public class GrpcToLogStreamGateway extends GatewayGrpc.GatewayImplBase implemen
   public void activateJobs(
       final ActivateJobsRequest request,
       final StreamObserver<ActivateJobsResponse> responseObserver) {
-    final Long requestId = registerNewRequest(responseObserver);
+    executor.submit(
+        () -> {
+          final Long requestId = registerNewRequest(responseObserver);
 
-    prepareRecordMetadata()
-        .requestId(requestId)
-        .valueType(ValueType.JOB_BATCH)
-        .intent(JobBatchIntent.ACTIVATE);
+          prepareRecordMetadata()
+              .requestId(requestId)
+              .valueType(ValueType.JOB_BATCH)
+              .intent(JobBatchIntent.ACTIVATE);
 
-    final JobBatchRecord jobBatchRecord = new JobBatchRecord();
+          final JobBatchRecord jobBatchRecord = new JobBatchRecord();
 
-    jobBatchRecord.setType(request.getType());
-    jobBatchRecord.setWorker(request.getWorker());
-    jobBatchRecord.setTimeout(request.getTimeout());
-    jobBatchRecord.setMaxJobsToActivate(request.getMaxJobsToActivate());
+          jobBatchRecord.setType(request.getType());
+          jobBatchRecord.setWorker(request.getWorker());
+          jobBatchRecord.setTimeout(request.getTimeout());
+          jobBatchRecord.setMaxJobsToActivate(request.getMaxJobsToActivate());
 
-    writeCommandWithoutKey(recordMetadata, jobBatchRecord);
+          writeCommandWithoutKey(recordMetadata, jobBatchRecord);
+        });
   }
 
   @Override

--- a/src/test/java/io/camunda/zeebe/process/test/multithread/WorkerTest.java
+++ b/src/test/java/io/camunda/zeebe/process/test/multithread/WorkerTest.java
@@ -1,0 +1,50 @@
+package io.camunda.zeebe.process.test.multithread;
+
+import io.camunda.zeebe.client.ZeebeClient;
+import io.camunda.zeebe.client.api.response.ProcessInstanceEvent;
+import io.camunda.zeebe.process.test.assertions.BpmnAssert;
+import io.camunda.zeebe.process.test.extensions.ZeebeProcessTest;
+import io.camunda.zeebe.process.test.testengine.InMemoryEngine;
+import io.camunda.zeebe.process.test.util.Utilities;
+import io.camunda.zeebe.process.test.util.Utilities.ProcessPackLoopingServiceTask;
+import java.util.Collections;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+
+@ZeebeProcessTest
+public class WorkerTest {
+
+  private ZeebeClient client;
+  private InMemoryEngine engine;
+
+  @Test
+  public void testJobsCanBeProcessedAsynchronouslyByWorker() throws InterruptedException {
+    // given
+    client
+        .newWorker()
+        .jobType(ProcessPackLoopingServiceTask.JOB_TYPE)
+        .handler(
+            (client, job) -> {
+              client.newCompleteCommand(job.getKey()).send();
+            })
+        .open();
+
+    Utilities.deployProcess(client, ProcessPackLoopingServiceTask.RESOURCE_NAME);
+    final Map<String, Object> variables =
+        Collections.singletonMap(ProcessPackLoopingServiceTask.TOTAL_LOOPS, 3);
+
+    // when
+    final ProcessInstanceEvent instanceEvent =
+        Utilities.startProcessInstance(
+            engine, client, ProcessPackLoopingServiceTask.PROCESS_ID, variables);
+
+    // then
+    BpmnAssert.assertThat(instanceEvent).isStarted();
+    // TODO: Idle state monitor does not work in this case.
+    //  Might be fixed when switching to the zeebe built-in idle state monitor
+    Thread.sleep(1000);
+    BpmnAssert.assertThat(instanceEvent)
+        .hasPassedElement(ProcessPackLoopingServiceTask.ELEMENT_ID, 3)
+        .isCompleted();
+  }
+}

